base/uk.ac.stfc.isis.ibex.opis/check_opi_format.py

```diff
@@ -0,0 +1,272 @@
+import os
+import sys
+from lxml import etree
+
+
+class CheckOpiFormat:
+
+    # Directory to iterate through
+    root_directory = r"C:\Instrument\Dev\ibex_gui\base\uk.ac.stfc.isis.ibex.opis\resources"
+
+    # Files ending with .opi are parsed:
+    file_extension = r".opi"
+
+    # If a word contains any of the following, the whole word will be ignored
+    ignore = \
+        ["$", "&", "#", '"', "'", "(", ")", "OPI", "PSU", "HW", "Hz", "LED", "A:", "B:", "C:", "D:"]
+
+    # When checking capitalisation will ignore words <= this length
+    # This is to avoid
+    ignore_short_words_limit = 3
+
+    def __init__(self):
+        self.errors = []
+
+    @staticmethod
+    def get_tree(filepath):
+        with open(filepath) as opi_file:
+            parser = etree.XMLParser(remove_blank_text=True)
+            return etree.parse(opi_file, parser)
+
+    def file_iterator(self):
+        for filename in os.listdir(self.root_directory):
+            if filename.endswith(self.file_extension):
+                filepath = os.path.join(self.root_directory, filename)
+                yield filepath
+
+    def check_plot_area_background(self, root):
```

Comment:
Why are you checking for plot area background, but not e.g. general foreground / background color?